src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -558,6 +565,15 @@ def execute_mock_plan(
         return result_df.where(result_df.notna(), None)  # Swap np.nan with None
     if isinstance(source_plan, MockFileOperation):
         return execute_file_operation(source_plan, analyzer)
+    if isinstance(source_plan, SnowflakeCreateTable):
+        table_registry = analyzer.session._conn.table_registry
+        res_df = execute_mock_plan(source_plan.query)
+        return table_registry.write_table(
+            source_plan.table_name, res_df, source_plan.mode
+        )
+    if isinstance(source_plan, UnresolvedRelation):
```

Comment:
Just curious: when will we need to deal with an `UnresolvedRelation`?

Reply:
It is only used for expressions like `session.table(<name>)`, I added a one line documentation in [src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py](https://github.com/snowflakedb/snowpark-python/pull/883/files#diff-19e42918750e256bcab12c18832dd863ebd4a9ce48978f91c1f7f272c0d4dda5)